src/__tests__/fixtures/schema/vacationRentals.ts

```diff
@@ -67,19 +67,22 @@ export class VacationRentals {
 	@Field({ elements: TigrisDataTypes.OBJECT, default: undefined })
 	reviews: Array<Object>;
 
-	@Field({ default: FieldDefaults.TIME_NOW })
+	@Field({ default: Generated.NOW })
+	availableSince: Date;
+
+	@Field({ default: Generated.TIMESTAMP })
 	lastSeen: Date;
 
-	@Field({ default: FieldDefaults.TIME_CREATED_AT })
+	@Field({ default: Generated.CREATED_AT })
```

Comment:
do we need to have CREATED_AT and UPDATED_AT as part of the defaults or should we have something like,
`{timestamp: Generated.CREATED_AT }` and `{timestamp: Generated. UPDATED_AT }`

Reply:
Because its in model, user can always override, in that case tagging it as default makes sense that we will only backfill when its missing. Managing types gets difficult otherwise if user has both a default and a timestamp in the field:

```ts
@Field({ default: new Date("01-01-1980"), timestamp: Generated.CREATED_AT })
myField: Date;
```

Reply:
The reason for separating it from default is that user can have both. As you've shown in your example as well. Assume that you only need to know the updatedAt of a row but you need to default in case the updatedAt is missing. In that case, you can have,
```
@Field({ default: now(), timestamp: Generated.Updated_At }) // or any default time.
```
If this is part of the default then there is no way for the user to set the default of the field.